test/libsolidity/Imports.cpp

```diff
@@ -186,6 +186,17 @@ BOOST_AUTO_TEST_CASE(remappings_ensure_order_doesnt_matter_in_reference)
 	BOOST_CHECK(c.compile());
 }
 
+BOOST_AUTO_TEST_CASE(remappings_reference_a_directory_tree)
+{
+	CompilerStack c;
+	c.setRemappings(vector<string>{"@foo/0.0.1/:s=s_1.4.6", "@bar/0.0.1/:s=s_1.4.7", "foo/0.0.1/=Foo", "bar/0.0.1/=Bar"});
```

Comment:
yea I added a separate test for that (and it fails as I expected it to), there's a problem with the ordering in the current setup.